### ISSUE TITLE
shfmt terminates command with both '&' and ';'

### ISSUE THREAD

**Issue by mikeBashStuff** [mvdan/sh#528]
Problematic scenario looks like the following:

```
# shellcheck-static <(shfmt-3.0.2 "${shfmt_cmdline[@]}" ./t)

In /dev/fd/63 line 3:
for foo; do "$foo" &;done
                    ^-- SC1045: It's not 'foo &; bar', just 'foo & bar'.

# shfmt_cmdline+=(-d)
# shfmt-3.0.2 "${shfmt_cmdline[@]}" ./t
--- ./t.orig
+++ ./t
@@ -1,4 +1,5 @@
 #!/bin/bash

-for foo; do "$foo" & done; wait
+for foo; do "$foo" &;done
+wait

# echo shfmt-3.0.2 "${shfmt_cmdline[@]}"
shfmt-3.0.2 -i 0 -bn -ci -ln bash -d
#

```
Where, unfortunately, this generates a syntax error, leaving the potential script completely unusable.

Moreover, the `shfmt` fails in the same manner even when call to `wait` is moved onto its own line. The only way to avoid this is by not keeping the `for` loop as a one-liner, i.e., to move its `done` keyword to a new line.

Any hints, comments would be appreciated. :)

**Comment by mvdan** [comment]
This looks like a duplicate of https://github.com/mvdan/sh/issues/520; if I try master, it works fine. It also roundtrips just fine too:

```
$ shfmt <<<'for foo; do "$foo" & done; wait'
for foo; do "$foo" & done
wait
$ shfmt <<<'for foo; do "$foo" & done; wait' | shfmt
for foo; do "$foo" & done
wait
```

3.1.0 should come out this week. If you don't want to wait, use master - it's pretty stable, and I use it all the time.

**Comment by mikeBashStuff** [comment]
@mvdan Thanks for the feedback! In my case, the actual release would be preferable. :) Some linux distros ship with golang packages that, for some reason, won't build the `shfmt` (case and point, `go1.11.6 ` shipped with Debian 10) - so for me, this introduces another dependency which I need to resolve by building `go` separately (and the `devel` build of `go` actually works with `shfmt` without any hiccups). 

Long story short, I am perfectly fine with waiting for the 3.1.0. Thanks again! :)

**Comment by mvdan** [comment]
Makes sense. Yeah, master and the latest releases generally require a somewhat recent version of Go, released in the past 6-12 months.
